tools/mtouch/errors.resx

```diff
@@ -0,0 +1,2623 @@
+		</comment>
+	</data>
+	
+	<data name="MM1403" xml:space="preserve">
+		<value>{0} {1} could not be found. Target framework '{2}' is unusable to package the application.
+		</value>
+		<comment>
+		</comment>
+	</data>
+	
+	<data name="MM1404" xml:space="preserve">
+		<value>Target framework '{0}' is invalid.
+		</value>
+		<comment>
+		</comment>
+	</data>
+	
+	<data name="MM1405" xml:space="preserve">
+		<value>useFullXamMacFramework must always target framework .NET 4.5, not '{0}' which is invalid.
+		</value>
+		<comment>
+		</comment>
+	</data>
+	
+	<data name="MM1407" xml:space="preserve">
+		<value>Mismatch between Xamarin.Mac reference '{0}' and target framework selected '{1}'.
+		</value>
+		<comment>
+		</comment>
+	</data>
+	
+	<data name="MM1501" xml:space="preserve">
+		<value>Can not resolve reference: {0}
+		</value>
+		<comment>
+		</comment>
+	</data>
+	
+	<data name="MM2006" xml:space="preserve">
+		<value>Native library '{0}' was referenced but could not be found.
+		</value>
+		<comment>
+		</comment>
+	</data>
+	
+	<data name="MM2007" xml:space="preserve">
+		<value>Xamarin.Mac Unified API against a full .NET framework does not support linking SDK or All assemblies. Pass either the `-nolink` or `-linkplatform` flag.
+		</value>
+		<comment>
+		</comment>
+	</data>
+	
+	<data name="MM2009" xml:space="preserve">
+		<value>  Referenced by {0}.{1}
+		</value>
+		<comment>
+		</comment>
+	</data>
+	
+	<data name="MM2012" xml:space="preserve">
+		<value> Only first {0} of {1} \"Referenced by\" warnings shown.
+		</value>
+		<comment>
+		</comment>
+	</data>
+	
+	<data name="MM2013" xml:space="preserve">
+		<value>Failed to resolve the reference to \"{0}\", referenced in \"{1}\". The app will not include the referenced assembly, and may fail at runtime.
+		</value>
+		<comment>
+		</comment>
+	</data>
+	
+	<data name="MM2103" xml:space="preserve">
+		<value>Error processing assembly '{0}': {1}
+		</value>
+		<comment>
+		</comment>
+	</data>
+	
+	<data name="MM2106" xml:space="preserve">
+		<value>Could not optimize the call to BlockLiteral.{2} in {0} at offset {1} because the previous instruction was unexpected ({3})
+		</value>
+		<comment>
+		</comment>
+	</data>
+	
+	<data name="MM2106_A" xml:space="preserve">
+		<value>Could not optimize the call to BlockLiteral.{2} in {0} at offset {1} because could not determine the type of the delegate type of the first argument (instruction: {3})
+		</value>
+		<comment>
+		</comment>
+	</data>
+	
+	<data name="MM2106_B" xml:space="preserve">
+		<value>Could not optimize the call to BlockLiteral.{2} in {0} because the type of the value passed as the first argument (the trampoline) is {1}, which makes it impossible to compute the block signature.
+		</value>
+		<comment>
+		</comment>
+	</data>
+	
+	<data name="MM2106_C" xml:space="preserve">
+		<value>Could not optimize the call to BlockLiteral.SetupBlock in {0} at offset {1} because no [UserDelegateType] attribute could be found o
```

Comment:
I think the messages should be sorted by number only (so that you get the MT*, MM*, and MX* messages all together), because a very common question is `What's the next available error number`? If the messages are split in three sections, we'd have to look in three sections every time. Half the time (at least) someone would forget, and try to use a number that's already in use. Having all the messages ordered by number alone should make that much less probable.

Reply:
There are some cases where the same error code being used for different error messages in mtouch and mmp. Do we want to handle these differently?

Example:

`MT0001: '-devname' was provided without any device-specific action`

`MM0001: This version of Xamarin.Mac requires Mono {0} (the current Mono version is {1}). Please update the Mono.framework from http://mono-project.com/Downloads`

Reply:
No, just put them next to each other in the file.